plugins/emoji/plugin.js

```diff
@@ -79,26 +152,466 @@
+					selector: 'input',
+					event: 'click',
+					listener: ( function() {
+						var inputNumber;
+						return function() {
+							if ( inputNumber === undefined ) {
+								inputNumber = getItemIndex( blockObject._.getItems().toArray(), blockElement.findOne( 'input' ) );
+							}
+							blockObject._.markItem( inputNumber );
+						};
+					} )()
+				} );
+				return '<label class="cke_emoji-search"><img src="' + escapeString( loupeUrl ) +
+					'" /><input placeholder="' + escapeString( lang.searchPlaceholder ) +
+					'" type="search" _cke_focus="1"></label>';
+			}
+
+			function createEmojiListBlock() {
+				listeners.push( {
+					selector: '.cke_emoji-outer_emoji_block',
+					event: 'scroll',
+					listener: ( function() {
+						var buffer = CKEDITOR.tools.throttle( 150, refreshNavigationStatus );
+						return buffer.input;
+					} )()
+				} );
+
+				listeners.push( {
+					selector: '.cke_emoji-outer_emoji_block',
+					event: 'click',
+					listener: function( event ) {
+						if ( event.data.getTarget().data( 'cke-emoji-name' ) ) {
+							editor.execCommand( 'insertEmoji', { emojiText: event.data.getTarget().data( 'cke-emoji-symbol' ) } );
+						}
+					}
+				} );
+
+				listeners.push( {
+					selector: '.cke_emoji-outer_emoji_block',
+					event: 'mouseover',
+					listener: function( event ) {
+						updateStatusbar( event.data.getTarget() );
+					}
+				} );
+
+				listeners.push( {
+					selector: '.cke_emoji-outer_emoji_block',
+					event: 'keyup',
+					listener: function() {
+						updateStatusbar( blockObject._.getItems().getItem( blockObject._.focusIndex ) );
+					}
+				} );
+
+				return '<div class="cke_emoji-outer_emoji_block">' + getEmojiSections() + '</div>';
+			}
+
+			function createStatusBar() {
+				return '<div class="cke_emoji-status_bar">' +
+					'<div class="cke_emoji-status_icon"></div>' +
+					'<div class="cke_emoji-status_description"><p class="cke_emoji-status_description"></p><p class="cke_emoji-status_full_name"></p></div>' +
+					'</div>';
+			}
+
+			function getEmojiSections() {
+				return CKEDITOR.tools.array.reduce( GROUPS, function( acc, item ) {
+					return acc + getEmojiSection( item );
+				}, '' );
+			}
+
+			function getEmojiSection( item ) {
+				var groupName = escapeString( item.name ),
+					sectionName = escapeString( item.sectionName ),
+					group = getEmojiListGroup( groupName );
+
+				return '<section data-cke-emoji-group="' + groupName + '" ><h2 id="' + groupName + '">' + sectionName + '</h2><ul>' + group + '</ul></section>';
+			}
+
+			function getEmojiListGroup( groupName ) {
+				var emojiList = editor._.emoji.list,
+					emojiTpl = new CKEDITOR.template( '<li class="cke_emoji_item">' +
+					'<a draggable="false" data-cke-emoji-full-name="{id}" data-cke-emoji-name="{name}" data-cke-emoji-symbol="{symbol}" data-cke-emoji-group="{group}" ' +
+					'data-cke-emoji-keywords="{keywords}" title="{id}" href="#" _cke_focus="1">{symbol}</a>' +
```

Comment:
Title should contain a human-readable label of an emoji. This will improve accessibility.